src/payment/strategies/cybersource/cardinal-client.ts

```diff
@@ -0,0 +1,241 @@
+    CardinalSetupCompletedData,
+    CardinalSignatureValidationErrors,
+    CardinalSDK,
+    CardinalTriggerEvents,
+    CardinalValidatedAction,
+    CardinalValidatedData
+} from './index';
+
+export default class CardinalClient {
+    private _sdk?: Promise<CardinalSDK>;
+    private _cardinalEvent$: Subject<CardinalEventResponse>;
+    private _testMode?: boolean;
+
+    constructor(
+        private _scriptLoader: CardinalScriptLoader
+    ) {
+        this._cardinalEvent$ = new Subject<CardinalEventResponse>();
+    }
+
+    initialize(testMode: boolean): Promise<void> {
+        this._testMode = testMode;
+
+        return this._getClientSDK().then(() => Promise.resolve());
+    }
+
+    configure(clientToken: string): Promise<void> {
+        return this._getClientSDK().then(client => {
+            client.on(CardinalEventType.SetupCompleted, (setupCompletedData: CardinalSetupCompletedData) => {
+                this._resolveSetupEvent();
+            });
+
+            client.on(CardinalEventType.Validated, (data: CardinalValidatedData, jwt: string) => {
+                switch (data.ActionCode) {
+                    case CardinalValidatedAction.Success:
+                        this._resolveAuthorizationPromise(jwt);
+                        break;
+                    case CardinalValidatedAction.NoAction:
+                        if (data.ErrorNumber > 0) {
+                            this._rejectAuthorizationPromise(data);
+                        } else {
+                            this._resolveAuthorizationPromise(jwt);
+                        }
+                        break;
+                    case CardinalValidatedAction.Failure:
+                        data.ErrorDescription = 'User failed authentication or an error was encountered while processing the transaction';
+                        this._rejectAuthorizationPromise(data);
+                        break;
+                    case CardinalValidatedAction.Error:
+                        if (includes(CardinalSignatureValidationErrors, data.ErrorNumber)) {
+                            this._rejectSetupEvent();
+                        } else {
+                            this._rejectAuthorizationPromise(data);
+                        }
+                }
+            });
+
+            return new Promise<void>((resolve, reject) => {
+                this._cardinalEvent$
+                    .pipe(take(1), filter(event => event.step === CardinalPaymentStep.Setup))
+                    .subscribe((event: CardinalEventResponse) => {
+                        event.status ? resolve() : reject(new MissingDataError(MissingDataErrorType.MissingPaymentMethod));
+                    });
+
+                client.setup(CardinalInitializationType.Init, {
+                    jwt: clientToken,
+                });
+            });
+        });
+    }
+
+    runBindProcess(ccNumber: string): Promise<void> {
+        return this._getClientSDK().then(client => {
```

Comment:
As we can see at the screenshot, we need to do this in order to give back the context to checkout-sdk, I mean if we just throw an exception it will be caught by `Cardinal SDK (songbird)` and we won't be able to continue executing the correct flow.

![image](https://user-images.githubusercontent.com/36899206/60146444-395ca080-978f-11e9-9713-43db77f0d586.png)